src/System.Composition.Convention/src/System/Composition/Convention/ConventionBuilder.cs

```diff
@@ -368,7 +368,7 @@ private static bool IsDescendentOf(Type type, Type baseType)
             var ti = type.GetTypeInfo();
             var bti = baseType.GetTypeInfo();
 
-            if (ti.IsGenericTypeDefinition)
+            if (ti.IsGenericTypeDefinition || bti.IsGenericTypeDefinition)
```

Comment:
No problem, will do. Basically here the change includes the case where the derivedType is a closed generic and the baseType is an open generic